Crud-App

SachinKiragi opened the pull request:
CRUD App in C

A console program in C to manage user info. We can add, view, update, and delete users (ID, name, age), and all data is saved in a text file so nothing gets lost. It checks for invalid input and prevents duplicate IDs, keeping things simple.

Features
- Add, view, update, and delete users
- Input validation for IDs, names, and ages
- Prevents duplicate IDs
- Stores all data in a file for persistence

SachinKiragi committed:
Push Crud.c

SachinKiragi committed:
Improved Message After Updating And Deleting User

SachinKiragi committed:
Handled Case When User Enters String As Input For Choice Op In Main

SachinKiragi committed:
Added Message After Creating User

SachinKiragi committed:
Replced fflush with getc

SachinKiragi committed:
Added Line Spacing

SachinKiragi committed:
Bug Fix

SachinKiragi committed:
Bug Fix

SachinKiragi committed:
Replacing fflush After scanf in main fn

tejas0123 commented on crud.c:
```diff
@@ -0,0 +1,263 @@
+
+int createFile(){
+    FILE *filePtr = fopen(fileName, "a");
+    if(filePtr!=NULL){
```
Operators and operands should be separated by whitespace

tejas0123 commented on crud.c:
```diff
@@ -0,0 +1,263 @@
+        printf("\nError Occured While Storing User Data\n");
+}
+
+int doesUserExist(int id){
```
Explore if boolean data type exists in C

tejas0123 commented on crud.c:
```diff
@@ -0,0 +1,263 @@
+
+    int len = length(currUser.name);
+    
+    if(len <= 1){
```
Data validation can be done using a different function. Consider writing a function`int isUserValid(User user)` and return error codes for different cases

tejas0123 commented on crud.c:
```diff
@@ -0,0 +1,263 @@
+
+void addUser(){
+
+    FILE* filePtr = fopen(fileName, "a");
```
Do you need to open file here? Open the file only when you insert and read. Maybe in the appendUser function. File should not be open for reading in multiple places. There will be risks of memory leaks

tejas0123 commented on crud.c:
```diff
@@ -0,0 +1,263 @@
+
+    User currUser = {.id=-1};
+
+    printf("Enter id (Number): ");
```
Input from the user can be taken in a different function and user object should be passed to the addUser function. It ensures that a function has single responsibility

SachinKiragi committed:
Refactored CRUD app: improved modularity, readability, and error handling

SachinKiragi commented:
Hello Sir!
I have updated the CRUD application as per the previous feedback:
- Improved code readability and spacing (Added space b/w Operands and Operator)
- Used bool instead of int for logical functions
- Added enum-based error handling
- Separated input and validation into modular functions
Please review the changes and let me know if any further improvements are needed.

SachinKiragi committed:
Comments Updated

tejas0123 commented on crud.c:
```diff
@@ -0,0 +1,333 @@
+
+int main(){
+
+    if(createFile() == 0){
```
Can we not use false here as we are using stdbool.h. Update 0 and 1 with true and false wherever used

tejas0123 commented:
Some small changes suggested. Update them and then it is good to go

SachinKiragi committed:
Merge branch 'master' into Crud-App
